File: gfdesignpatterns/Ruby/factory.rb
class Show
require 'date'
	def initialize()

		rand = Random.new()
		randomNum = rand.rand(20)
		@year = randomNum + 2015
		
		end
	
	def to_s
		"Name is: #{@name} Year is: #{@year} Type is: #{@type}"
	end
	
	def buildname()
		raise NotImplementedError, "#{self.class.type} does not implement buildName()"
	end
	
	
end

class Movie < Show
	def initialize()
		super
		@type = "Movie"
		@movieNames = ["Toy", "Story", "A", "Bug's", "Life", "2", "Monsters", "Inc.", "Finding", "Nemo", "The", "Incredibles", "Cars", "Ratatouille", "WALL-E", "Up", "3", "Brave", "University", "Inside", "Out"]

		rand = Random.new()
		repeatTimes = rand.rand(5)
		arraySize = @movieNames.length
		randomName = ""
		while repeatTimes>0
			repeatTimes=repeatTimes-1
			randomLocation = rand.rand(arraySize)
			randomName+=@movieNames[randomLocation]+" "
		end
		@name = randomName
	
	end
	
end
	
class Short < Show
	def initialize()
		super
		@type = "Short"
		@shortNames = ["The", "Adventures", "of", "Andre", "and", "Wally", "Luxo", "Jr", "Reds", "Dream", "Tin", "Toy", "Knick", "Knack", "Geris", "Game", "For", "Birds", "Boundin", "One", "Man", "Band", "Lifted", "Presto", "Partly", "Cloudy", "Day", "and", "Night", "La", "Luna", "Blue", "Umbrella", "Lava", "Sanjays", "Super", "Team", "Mike's", "New", "Car", "JackJack", "Attack", "Mater", "Ghostlight", "Your", "Friend", "Rat", "BURNE", "Dugs", "Special", "Mission", "George", "AJ", "Legend", "Mordu", "Party", "Central"]

		rand = Random.new()
		repeatTimes = rand.rand(5)
		arraySize = @shortNames.length
		randomName = ""
		while repeatTimes>0
			repeatTimes=repeatTimes-1
			randomLocation = rand.rand(arraySize)
			randomName+=@shortNames[randomLocation]+" "
		end
		@name = randomName
	
	end
	
end


class ShowFactory
	def initialize()
	end
	def create(type)
		case type	
			when 1
				return Short.new()
			else
				return Movie.new()
		end
	end
end

puts "Show 1"
factory = ShowFactory.new()
show1 = factory.create(1)
puts show1.to_s
puts "\nShow 2"	
show2 = factory.create(1)
puts show2.to_s
puts "\nShow 3"
show3 = factory.create(0)
puts show3.to_s
puts "\nShow 4"
show4 = factory.create(0)
puts show4.to_s	
		
		
		
		
		
		
		
		
		
		
		
		
		
		
		
		
		
		
		
		
		
		
		
		
		

File: gfdesignpatterns/gfdesignpatterns/Factory/factory.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace gfdesignpatterns.Factory
{
    public abstract class Show
    {
        protected List<string> movieNames = new List<string>{"Toy", "Story", "A", "Bug's", "Life", "2", "Monsters", "Inc.", "Finding", "Nemo", "The", "Incredibles", "Cars", "Ratatouille", "WALL-E", "Up", "3", "Brave", "University", "Inside", "Out"};
        protected List<string> shortNames = new List<string>{"The", "Adventures", "of", "André", "and", "Wally", "Luxo", "Jr.", "Red's", "Dream", "Tin", "Toy", "Knick", "Knack", "Geri's", "Game", "For", "Birds", "Boundin'", "One", "Man", "Band", "Lifted", "Presto", "Partly", "Cloudy", "Day", "&", "Night", "La", "Luna", "Blue", "Umbrella", "Lava", "Sanjay's", "Super", "Team", "Mike's", "New", "Car", "Jack-Jack", "Attack", "Mater", "Ghostlight", "Your", "Friend", "Rat", "BURN-E", "Dug's", "Special", "Mission", "George", "A.J.", "Legend", "Mor'du", "Party", "Central"};

        public abstract string type { get; }
        public abstract string name { get; }

        public string year
        {
            get
            {
                DateTime start = new DateTime(2015, 1, 1);
                Random gen = new Random();
                int range = (20);
                var newdate= start.AddYears(gen.Next(range));
                return (gen.Next(range) + 2015).ToString();
            }

        }
    }

    public class Movie : Show
    {

        public override string type
        {
            get { return "Movie"; }
        }

        public override string name
        {
            get
            {
                Random rnd = new Random();
                int count = rnd.Next(1, 6);
                string result = "";
                for (int i = 0; i < count; i++)
                {
                    result += movieNames[rnd.Next(0, movieNames.Count() - 1)]+" ";
                }
                return result;
            }
        }
    }

    public class Short : Show
    {

        public override string type
        {
            get
            {
                return "Short";
            }

        }
        public override string name
        {
            get
            {
                Random rnd = new Random();
                int count = rnd.Next(1, 6);
                string result = "";
                for (int i = 0; i < count; i++)
                {
                    result += shortNames[rnd.Next(0, shortNames.Count() - 1)] + " ";
                }
                return result;
            }
        }
    }

    public static class ShowFactory
    {
        public static Show get(string type)
        {
            switch (type.ToLower())
            {
                case "1": return new Movie();
                default: return new Short();
            }
        }
    }


}